game/src/main/java/jake2/game/GameExportsImpl.java

```diff
@@ -1650,4 +1678,21 @@ public int getNumEdicts() {
         return num_edicts;
     }
 
+    @Override
+    public void fromPrevious(GameExports oldGame) {
+        GameExportsImpl oldGameImpl = (GameExportsImpl) oldGame;
+        // assert both client arrays are of the same size
```

Comment:
This is not clear.
Feels like there should be an assert here and then only one length should be checked in the `for` condition.